src/util/benchmark.cc

```diff
@@ -0,0 +1,190 @@
+{
+  uint8_t private_key_DSA[kovri::core::DSA_PRIVATE_KEY_LENGTH];
+  uint8_t public_key_DSA[kovri::core::DSA_PUBLIC_KEY_LENGTH];
+  uint8_t output_DSA[kovri::core::DSA_SIGNATURE_LENGTH];
+  LOG(info) << "--------DSA---------";
+  BenchmarkTest<kovri::core::DSAVerifier, kovri::core::DSASigner>(
+      Benchmark::BenchmarkCount,
+      private_key_DSA,
+      public_key_DSA,
+      output_DSA,
+      kovri::core::CreateDSARandomKeys);
+
+  uint8_t private_key_ECDSAP256[kovri::core::ECDSAP256_KEY_LENGTH];
+  uint8_t public_key_ECDSAP256[kovri::core::ECDSAP256_KEY_LENGTH/ 2];
+  uint8_t output_ECDSAP256[kovri::core::ECDSAP256_KEY_LENGTH];
+  LOG(info) << "-----ECDSAP256------";
+  BenchmarkTest<kovri::core::ECDSAP256Verifier, kovri::core::ECDSAP256Signer>(
+      Benchmark::BenchmarkCount,
+      private_key_ECDSAP256,
+      public_key_ECDSAP256,
+      output_ECDSAP256,
+      kovri::core::CreateECDSAP256RandomKeys);
+
+  LOG(info) << "-----ECDSAP384------";
+  uint8_t private_key_ECDSAP384[kovri::core::ECDSAP384_KEY_LENGTH];
+  uint8_t public_key_ECDSAP384[kovri::core::ECDSAP384_KEY_LENGTH / 2];
+  uint8_t output_ECDSAP384[kovri::core::ECDSAP384_KEY_LENGTH];
+  BenchmarkTest<kovri::core::ECDSAP384Verifier, kovri::core::ECDSAP384Signer>(
+      Benchmark::BenchmarkCount,
+      private_key_ECDSAP384,
+      public_key_ECDSAP384,
+      output_ECDSAP384,
+      kovri::core::CreateECDSAP384RandomKeys);
+
+  LOG(info) << "-----ECDSAP521------";
+  uint8_t private_key_ECDSAP521[kovri::core::ECDSAP521_KEY_LENGTH];
+  uint8_t public_key_ECDSAP521[kovri::core::ECDSAP521_KEY_LENGTH / 2];
+  uint8_t output_ECDSAP521[kovri::core::ECDSAP521_KEY_LENGTH];
+  BenchmarkTest<kovri::core::ECDSAP521Verifier, kovri::core::ECDSAP521Signer>(
+      Benchmark::BenchmarkCount,
+      private_key_ECDSAP521,
+      public_key_ECDSAP521,
+      output_ECDSAP521,
+      kovri::core::CreateECDSAP521RandomKeys);
+
+  LOG(info) << "-----EDDSA25519-----";
+  uint8_t private_key_EDDSA25519[kovri::core::EDDSA25519_PRIVATE_KEY_LENGTH];
+  uint8_t public_key_EDDSA25519[kovri::core::EDDSA25519_PUBLIC_KEY_LENGTH];
+  uint8_t output_EDDSA25519[ kovri::core::EDDSA25519_SIGNATURE_LENGTH];
+  BenchmarkTest<kovri::core::EDDSA25519Verifier, kovri::core::EDDSA25519Signer>(
+      Benchmark::BenchmarkCount,
+      private_key_EDDSA25519,
+      public_key_EDDSA25519,
+      output_EDDSA25519,
+      kovri::core::CreateEDDSARandomKeys);
+}
+
+Benchmark::Benchmark() : m_Desc("Options")
+{
+  m_Desc.add_options()("help,h", "produce this help message")
+     ("test,t", bpo::bool_switch()->default_value(false), "all tests");
+}
+/// @brief parse options and perform action
+bool Benchmark::Impl(const std::string& cmd_name,
+    const std::vector<std::string>& args)
+{
+  bpo::variables_map vm;
+  try
+    {
+      bpo::parsed_options parsed =
+          bpo::command_line_parser(args).options(m_Desc).run();
+      bpo::store(parsed, vm);
+      bpo::notify(vm);
+    }
+  catch (bpo::error& e)
```

Comment:
Please `catch (....)` and use the exception dispatcher. It will catch `boost::program_options::error`.